main.go

```diff
@@ -284,17 +237,78 @@ func main() {
 		os.Exit(1)
 	}
 
+	clientset, clientErr := kubernetes.NewForConfig(mgr.GetConfig())
+	if err != nil {
+		setupLog.Error(clientErr, "failed to create kubernetes clientset")
+	}
+
+	reviewer := rbac.NewReviewer(clientset)
 	ctx := ctrl.SetupSignalHandler()
-	err = addDependencies(ctx, mgr, cfg, v)
+
+	// builds the operator's configuration
+	ad, err := autodetect.New(restConfig, ctx, reviewer)
 	if err != nil {
-		setupLog.Error(err, "failed to add/run bootstrap dependencies to the controller manager")
+		setupLog.Error(err, "failed to setup auto-detect routine")
 		os.Exit(1)
 	}
-	clientset, clientErr := kubernetes.NewForConfig(mgr.GetConfig())
+
+	rbacAvailable, err := ad.RBACPermissions()
 	if err != nil {
-		setupLog.Error(clientErr, "failed to create kubernetes clientset")
+		createRBACPermissions = false
+		setupLog.Info("error checking if the operator has permissions to create RBAC resources", "reason", err)
+	} else if rbacAvailable == autoRbac.NotAvailable {
+		createRBACPermissions = false
+		setupLog.Info("The operator has not permissions to create RBAC resources, skipping.")
+	} else {
+		createRBACPermissions = true
+	}
+
+	cfg := config.New(
+		config.WithLogger(ctrl.Log.WithName("config")),
+		config.WithVersion(v),
+		config.WithCollectorImage(collectorImage),
+		config.WithCreateRBACPermissions(createRBACPermissions),
```

Comment:
we don't need this option anymore and should instead use the autodetect's RBACPermissions() method